### PR TITLE
Upgrade django to 1.10

### PR DESCRIPTION
**Pull request by umgelurgel**
TODO:
* ~~`TestCFP` failures under Django 1.9 caused by `Sekizai` library.~~
* ~~`Manager` classes redefining `__getattr__` causing a runtime error under Django 1.10~~
* ~~Test failures under 1.10~~

All issues fixed, changes ready for review.

### CODE REVIEW

File: assopy/admin.py [hunk 1]
```diff
@@ -197,7 +197,7 @@ def vouchers(self, request):
                 .filter(conference=dsettings.CONFERENCE_CONFERENCE, payment_type='v'),
         }
         return render_to_response(
-            'admin/assopy/order/vouchers.html', ctx, context_instance=template.RequestContext(request))
+            'admin/assopy/order/vouchers.html', ctx)
```
Comment (artcz) [inline assopy/admin.py]: lets replace that `render_to_response` with `render()` or `TemplateResponse`.

File: assopy/admin.py [hunk 2]
```diff
@@ -182,12 +182,12 @@ def _invoice(self, o):
     def get_urls(self):
         urls = super(OrderAdmin, self).get_urls()
         f = self.admin_site.admin_view
-        my_urls = patterns('',
+        my_urls = [
             url(r'^invoices/$', f(self.edit_invoices), name='assopy-edit-invoices'),
```
Comment (artcz) [inline assopy/admin.py]: that `f` is nice and short, but calling it 'admin_view' would be better.

File: assopy/admin.py [hunk 3]
```diff
@@ -208,7 +208,7 @@ def vouchers_fare(self, request, conference, fare):
             'items': items,
         }
         return render_to_response(
-            'admin/assopy/order/vouchers_fare.html', ctx, context_instance=template.RequestContext(request))
+            'admin/assopy/order/vouchers_fare.html', ctx)
```
Comment (artcz) [inline assopy/admin.py]: ditto

File: assopy/admin.py [hunk 4]
```diff
@@ -248,7 +248,7 @@ class FormPaymentDate(forms.Form):
             'form': form,
             'ids': request.GET.get('id'),
         }
-        return render_to_response('assopy/admin/edit_invoices.html', ctx, context_instance=template.RequestContext(request))
+        return render_to_response('assopy/admin/edit_invoices.html', ctx)
```
Comment (artcz) [inline assopy/admin.py]: ditto

File: assopy/models.py [hunk 5]
```diff
@@ -458,37 +458,30 @@ def _applyToTotal(self, total, guard):
...
-                qs = qs.filter(price__gt=0)
-            t = qs.aggregate(t=models.Sum('price'))['t']
-            return t if t is not None else 0
+    def usable(self, include_admin=False):
+        """
+        restituisce tutti gli ordini "usabili", cioè tutti gli ordini con
+        metodo bonifico (a prescindere se risultano pagati o meno) e tutti
+        gli ordini con metodo paypal (o cc) completati.
+        """
+        qs = self.filter(models.Q(method='bank')|models.Q(method__in=('cc', 'paypal'), _complete=True))
```
Comment (artcz) [inline assopy/models.py]: might be more readable to go with `Q` directly. (ie. models.Q -> Q)

File: assopy/models.py [hunk 6]
```diff
@@ -668,7 +661,7 @@ class Order(models.Model):
         null=True,
     )
 
-    objects = OrderManager()
+    objects = OrderQuerySet.as_manager()
```
Comment (artcz) [inline assopy/models.py:664]: :+1:

File: assopy/urls.py [hunk 7]
```diff
@@ -16,6 +16,7 @@
 )
 
 from assopy.forms import LoginForm, SetPasswordForm
+from assopy import views as assopy_views
```
Comment (artcz) [inline assopy/urls.py]: unless there's some conflict here, we could go with just `views` since it's already inside `assopy` app.

File: common/decorators.py [hunk 8]
```diff
@@ -70,10 +69,8 @@ def wrapper(request, *args, **kw):
             if isinstance(output, dict):
                 if request.is_ajax():
                     tpl = ('%s_body%s' % os.path.splitext(tpl), tpl)
-                return render_to_response(tpl,
-                                          output,
-                                          RequestContext(request),
-                                          content_type=ct)
+
+                return render(request, tpl, output, content_type=ct)
```
Comment (artcz) [inline common/decorators.py:73]: good change. for future I'd also consider removing that decorator altogether, and instead handle the output format and template name in the view function/class directly. (for {read,debug}ability)

File: common/jsonify.py [hunk 9]
```diff
@@ -14,6 +13,9 @@ def default(self, obj):
             return obj.strftime('%H:%M')
         elif isinstance(obj, set):
             return list(obj)
+        elif callable(obj):
```
Comment (artcz) [inline common/jsonify.py:16]: smart. do we serialise user object to json anywhere?

File: conference/admin.py [hunk 10]
```diff
@@ -49,7 +49,7 @@ def _attendee_stats(self, o):
 
     def get_urls(self):
         v = self.admin_site.admin_view
```
Comment (artcz) [inline conference/admin.py]: at least here it' a `v` and not `f` :)

File: conference/admin.py [hunk 11]
```diff
@@ -119,7 +118,7 @@ def schedule_view_track(self, request, cid, sid, tid):
         return render_to_response(
```
Comment (artcz) [inline conference/admin.py]: ditto render_to_response -> TemplateResponse

File: formstyle/templatetags/formstyle.py [hunk 12]
```diff
@@ -24,10 +26,10 @@ def form_field(field, classes=None):
         classes = ' '.join(classes)
     classes += " field {} {}".format(widget_name.lower(), field_name.lower())
 
-    ctx = template.Context({
+    ctx = {
         'field': field,
         'classes': classes
-    })
+    }
     return tpl.render(ctx)
```
Comment (artcz) [inline formstyle/templatetags/formstyle.py]: that looks like it could be replaced  by `render_to_string`

File: formstyle/templatetags/formstyle.py [hunk 13]
```diff
@@ -46,8 +48,8 @@ def form_errors(form, classes=None):
         classes = ' '.join(classes)
     classes += form_name
 
-    ctx = template.Context({
+    ctx = {
         'form': form,
         'classes': classes
-    })
+    }
     return tpl.render(ctx)
```
Comment (artcz) [inline formstyle/templatetags/formstyle.py]: ditto (render_to_string)

File: pycon/settings.py [hunk 14]
```diff
@@ -335,6 +335,7 @@
 
     'markitup',
     'cms_utils',
+    'social_django',
```
Comment (artcz) [inline pycon/settings.py:338]: what's social_django?

**Review by artcz — APPROVED** [review]

**Review by artcz — CHANGES_REQUESTED** [review]
Looks OK, I've added comments about retiring render_to_response, and favoring render_to_string over Template and Context where applicable. Also some notes for the future (for example retiring render_to_template) which should be done in a separate PR after this one is merged.

**Comment by artcz** [comment]
(sorry for the mess with the comments, I clicked a wrong review button. :))

File: common/jsonify.py (hunk 9 above)
Comment (umgelurgel) [reply common/jsonify.py:16]: This is actually for `User.is_authenticated` that used to be a function and from 1.10 is a property that returns a `BoolCallable` - which works as a property and a method, so it's a transparent change when used in Python, but breaks serialization. More info here: https://docs.djangoproject.com/en/2.1/releases/1.10/#using-user-is-authenticated-and-user-is-anonymous-as-methods

File: common/decorators.py (hunk 8 above)
Comment (umgelurgel) [reply common/decorators.py:73]: I agree with removing the decorator - it is confusing in terms of program flow.

File: pycon/settings.py (hunk 14 above)
Comment (umgelurgel) [reply pycon/settings.py:338]: I think it's part of `python-social-auth` that's been refactor the the separate package - the whole `python-social-auth` module is now depracated and has been split into a number of smaller, self contained packages. The `tests.test_cfp.TestCFP testMethod=test_665_edit_with_other_in_django_admin` test would fail without this change to the `settings.

**Comment by umgelurgel** [comment]
Thanks for the review @artcz - all comments have been addressed and questions answered! Let me know if I've missed anything.

File: assopy/models.py [hunk 15]
```diff
@@ -458,37 +458,30 @@ def _applyToTotal(self, total, guard):
...
-            if include_admin:
-                qs = qs.filter(method='admin')
-            return qs
+    def usable(self, include_admin=False):
+        """
+        restituisce tutti gli ordini "usabili", cioè tutti gli ordini con
+        metodo bonifico (a prescindere se risultano pagati o meno) e tutti
+        gli ordini con metodo paypal (o cc) completati.
+        """
+        qs = self.filter(Q(method='bank') | Q(method__in=('cc', 'paypal'), _complete=True))
```
Comment (artcz) [inline assopy/models.py:474]: doesn't it basically mean `.filter(method__in=['bank', 'cc', 'paypal'], _complete=True)`?

and by extension `.filter(method_in=NON_ADMIN_PAYMENT_METHODS, _complete=True)` (where NON_ADMIN... is not defined anywhere yet, but it adds some semantics to this line.

File: common/jsonify.py (hunk 9 above)
Comment (artcz) [reply common/jsonify.py:16]: cool, but where is it used? sessions maybe?

File: assopy/admin.py [hunk 16]
```diff
@@ -495,7 +498,7 @@ def clean_coupon(self):
             'user': user,
             'form': form,
         }
-        return render_to_response('admin/auth/user/new_order.html', ctx, context_instance=template.RequestContext(request))
+        return TemplateResponse('admin/auth/user/new_order.html', ctx)
```
Comment (artcz) [inline assopy/admin.py]: one of the API changes is that TemplateResponse takes request as first argument, I think it needs to be passed here as well. (and all other TemplateResponse uses)

Comment (umgelurgel) [reply assopy/admin.py]: Thanks for catching this, got it mixed up with `SimpleTemplateResponse`. I guess admin views are not covered in tests that's why this didn't raise any flags in CI. Will address shortly.

File: common/jsonify.py (hunk 9 above)
Comment (umgelurgel) [reply common/jsonify.py:16]: The template causing the issue was `templates/p3/base.html` in the following lines:
```
var user = {
            'authenticated': {{ user.is_authenticated|json_ }}
        }
```
The user object seems to be used in `schedule.js` file.

File: assopy/models.py (hunk 15 above)
Comment (umgelurgel) [reply assopy/models.py:474]: Actually, taking it a level of abstraction higher, as the method is buggy and the implementation does not match the implementation (_Returns all "usable" orders, ie all orders with bank transfer method  of whether they are paid or not) and all  with paypal (or cc) method completed._) - this queryset method is not used anyhwere. I'll just remove it.

File: common/jsonify.py (hunk 9 above)
Comment (artcz) [reply common/jsonify.py:16]: Let's keep it that way then, something to address in the future.

File: assopy/admin.py (hunk 16 above)
Comment (artcz) [reply assopy/admin.py]: maybe we should also add basic HTTP 200 tests/stubs for those views, while there's no request there, so make sure they all fail, so that they work after this change? :)

Comment (umgelurgel) [reply assopy/admin.py]: I'm up for doing that - the only concern being the PR getting too large (the diff is at ~600 lines right now) and I won't be able to do it until the weekend/early next week.

File: assopy/models.py (hunk 15 above)
Comment (artcz) [reply assopy/models.py:474]: LGTM

File: assopy/admin.py (hunk 16 above)
Comment (artcz) [reply assopy/admin.py]: take your time, don't need to rush this upgrade. :)
as for lines – yeah, unfortunately. 
we could try merging that separately before this PR (the testsuite should still be compatible with both "versions"), but at the same time, I don't mind one big PR either – it's more important to include tests for the code that's being changed.

**Comment by umgelurgel** [comment]
A bit of a large commit, the last one - while writing tests, I've realised that the factories in `assopy` are set up in a suboptimal and confusing way. I've fixed most of them, but I've realised the other factories are messed up, too - that's going to be out of the scope of this PR though. 
tl;dr - comments addressed :)

**Review by artcz — APPROVED** [review]
LGTM. Great work. :)
